Add registries warning to registries.conf

TomSweeneyRedHat opened the pull request:
Add a warning about fully specifying registries
to the search list.

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

TomSweeneyRedHat commented:
@rhatdan PTAL

rhatdan commented on registries.conf:
```diff
@@ -8,6 +8,11 @@
 #
 # Registries to search for images that are not fully-qualified.
 # i.e. foobar.com/my_image:latest vs my_image:latest
+# NOTE: pulling an image that is not fully qualified, i.e. one that
```
I think this needs more description.  Let's use a google doc to start discussion.

mtrmac commented:
I’m perfectly fine with adding a warning here, but note that the vast majority of users of short names will never read this config file, so warning _only_ here would not be sufficient.

rhatdan commented:
Agreed, we also need to add this to container-registires.conf.md man page.

TomSweeneyRedHat commented:
As the comments in the googledoc have died down, I've just updated this PR with the latest changes from there.  I did make some line feed changes for it to fit under 80 char, but left the wording intact.

TomSweeneyRedHat commented:
And I think I made it past the validation test finally.

rhatdan commented:
Can you add the same information into the man page for container-registries.conf.5.md

vrothberg commented:
Once we consolidated the search-code into c/image, I'd love to throw a warning in the logs for that as well.

TomSweeneyRedHat commented:
I've added to the man page too.  FWIW, I made one small change, at the start of the second paragraph we had "Note that it is recommended".  I changed that to "It is recommended".  We'd used the word in another place or two and it just sounded klunky to me.

rhatdan commented:
LGTM
